File: backend/src/main/java/com/inventario/service/saas/PlanEntitlementCodes.java
package com.inventario.service.saas;

/**
 * Códigos de módulos/servicios del catálogo SaaS (matriz de planes).
 * Valores alineados con contrato de producto; no exponer como enum JPA.
 */
public final class PlanEntitlementCodes {

    private PlanEntitlementCodes() {}

    public static final String INVENTARIO_BASICO = "inventario_basico";
    public static final String CATEGORIAS = "categorias";
    public static final String MOVIMIENTOS_BASICOS = "movimientos_basicos";
    public static final String CONSULTA_STOCK = "consulta_stock";
    public static final String MULTI_BODEGA = "multi_bodega";
    public static final String TRANSFERENCIAS = "transferencias";
    public static final String AJUSTES_INVENTARIO = "ajustes_inventario";
    public static final String PROVEEDORES = "proveedores";
    public static final String USUARIOS = "usuarios";
    public static final String CONFIGURACION_EMPRESA = "configuracion_empresa";
    public static final String REPORTES_BASICOS = "reportes_basicos";
    public static final String HISTORIAL_MOVIMIENTOS = "historial_movimientos";
    public static final String ROLES_AVANZADOS = "roles_avanzados";
    public static final String REPORTES_AVANZADOS = "reportes_avanzados";
    public static final String AUDITORIA = "auditoria";
    public static final String MULTI_SEDE = "multi_sede";
    public static final String INTEGRACIONES = "integraciones";
    public static final String SOPORTE_PRIORITARIO = "soporte_prioritario";
}

File: src/app/core/plan-entitlement-codes.ts
/** Códigos alineados con `PlanEntitlementCodes` del backend (matriz de planes). */
export const PlanEntitlementCodes = {
  inventario_basico: 'inventario_basico',
  categorias: 'categorias',
  movimientos_basicos: 'movimientos_basicos',
  consulta_stock: 'consulta_stock',
  multi_bodega: 'multi_bodega',
  transferencias: 'transferencias',
  ajustes_inventario: 'ajustes_inventario',
  proveedores: 'proveedores',
  usuarios: 'usuarios',
  configuracion_empresa: 'configuracion_empresa',
  reportes_basicos: 'reportes_basicos',
  historial_movimientos: 'historial_movimientos',
  reportes_avanzados: 'reportes_avanzados',
  roles_avanzados: 'roles_avanzados',
  auditoria: 'auditoria',
  multi_sede: 'multi_sede',
  integraciones: 'integraciones',
  soporte_prioritario: 'soporte_prioritario'
} as const;

export type PlanEntitlementCode = (typeof PlanEntitlementCodes)[keyof typeof PlanEntitlementCodes];
